packages/tools/src/tools/displayTools/Labelmap/addLabelmapToElement.ts

```diff
@@ -31,27 +33,45 @@ async function addLabelmapToElement(
   const immediateRender = false;
   const suppressEvents = true;
 
-  // Todo: Right now we use MIP blend mode for the labelmap, since the
-  // composite blend mode has a non linear behavior regarding fill and line
-  // opacity. This should be changed to a custom labelmap blendMode which does
-  // what composite does, but with a linear behavior.
-  const volumeInputs: Types.IVolumeInput[] = [
-    {
-      volumeId,
-      actorUID: segmentationRepresentationUID,
-      visibility,
-      blendMode: Enums.BlendModes.MAXIMUM_INTENSITY_BLEND,
-    },
-  ];
+  if (isVolumeSegmentation(labelMapData)) {
+    // Todo: Right now we use MIP blend mode for the labelmap, since the
+    // composite blend mode has a non linear behavior regarding fill and line
+    // opacity. This should be changed to a custom labelmap blendMode which does
+    // what composite does, but with a linear behavior.
+    const volumeInputs: Types.IVolumeInput[] = [
+      {
+        volumeId: labelMapData.volumeId,
+        actorUID: segmentationRepresentationUID,
+        visibility,
+        blendMode: Enums.BlendModes.MAXIMUM_INTENSITY_BLEND,
+      },
+    ];
+
+    // Add labelmap volumes to the viewports to be be rendered, but not force the render
+    await addVolumesToViewports(
+      renderingEngine,
+      volumeInputs,
+      [viewportId],
+      immediateRender,
+      suppressEvents
+    );
+  } else {
+    const stackInputs: Types.IStackInput[] = [
+      {
+        imageId: viewport.getCurrentImageId(),
```

Comment:
This should mirror the volume one and use the stackseg:.... image id instead, otherwise you don't get the right metadata in the create image data.  If you need to include the current image id as well, that is certainly possible.